test/extended/builds/new_app.go

```diff
@@ -0,0 +1,41 @@
+package builds
+
+import (
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[builds][Slow] new-app with long name", func() {
```

Comment:
actually, yes, I think this needs to be an extended test.  I specifically want to ensure that the creation of an app with a long name succeeds to the point of application deployment - this is to guard against people adding labels containing the app name in future.